ompi/mca/coll/inter/coll_inter_allgather.c

```diff
@@ -87,12 +89,28 @@ mca_coll_inter_allgather_inter(const void *sbuf, int scount,
     }
     /* bcast the message to all the local processes */
     if ( rcount > 0 ) {
-	err = comm->c_local_comm->c_coll->coll_bcast(rbuf, rcount*rsize, rdtype,
-						    root, comm->c_local_comm,
-						    comm->c_local_comm->c_coll->coll_bcast_module);
-	if (OMPI_SUCCESS != err) {
-	    goto exit;
-	}
+        if ( OPAL_UNLIKELY(rcount*(size_t)rsize > INT_MAX) ) {
+            // This message would exceed the 'int count' paramater in
```

Comment:
I don't think this comment really reflects what the code is doing.

Reply:
I see. How about:
```
// Sending the message in the coll_bcast as "rcount*rsize" would exceed
// the 'int count' parameter in the coll_bcast() function. Instead broadcast
// the result in "rcount" chunks to the local group.
```

Reply:
FYI: I went ahead and pushed the updated comment. Let me know what you think and if other adjustments are needed.